### PR TITLE
Add reading password from attributes

### PR DESCRIPTION
**Pull request by 74wny0wl**
Allows uploading encrypted archives (ex. zip file encrypted with ZipCrypto)
via MWDB with password set as attribute

Note: 7z encrypted archives with AES (filenames encryption enabled) are not supported

### CODE REVIEW

**Comment by nazywam** [comment]
Hey, thanks for the PR!

What do you thinkg about doing it this way?
```python
attributes = task.get_payload("attributes", default={})
if not task_password and attributes.get("password"):
    self.log.info("Accepting password from attributes")
    task_password = attributes.get("password")[0]
```
It would allow us to get rid of that nasty `(attributes_password["password"] or [None])[0]`

**Comment by 74wny0wl** [comment]
> Hey, thanks for the PR!
> 
> What do you thinkg about doing it this way?
> 
> ```python
> attributes = task.get_payload("attributes", default={})
> if not task_password and attributes.get("password"):
>     self.log.info("Accepting password from attributes")
>     task_password = attributes.get("password")[0]
> ```
> 
> It would allow us to get rid of that nasty `(attributes_password["password"] or [None])[0]`

Yes, using an empty dict as the default result is much better :)

**Comment by 74wny0wl** [comment]
> Hey, thanks for the PR!
> 
> What do you thinkg about doing it this way?
> 
> ```python
> attributes = task.get_payload("attributes", default={})
> if not task_password and attributes.get("password"):
>     self.log.info("Accepting password from attributes")
>     task_password = attributes.get("password")[0]
> ```
> 
> It would allow us to get rid of that nasty `(attributes_password["password"] or [None])[0]`

Yes, it is much better :)

File: karton/archive_extractor/archive_extractor.py [hunk 1]
```diff
@@ -27,6 +27,11 @@ class ArchiveExtractor(Karton):
     def process(self, task: Task) -> None:
         sample = task.get_resource("sample")
         task_password = task.get_payload("password", default=None)
+        
+        attributes_password = task.get_payload("attributes", default=None)
+        if not task_password and attributes_password and "password" in attributes_password:
+            self.log.info("Taking password from attributes")
+            task_password = (attributes_password["password"] or [None])[0]
```
Comment (nazywam) [inline karton/archive_extractor/archive_extractor.py:34]:
```suggestion
        attributes = task.get_payload("attributes", default={})
        if not task_password and attributes.get("password"):
            self.log.info("Accepting password from attributes")
            task_password = attributes.get("password")[0]
```

**Review by nazywam — APPROVED** [review]

**Pull request merged**